demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -0,0 +1,998 @@
+                                                               print_as_warnings=self.print_ignored_errors,
+                                                               should_version_raise=should_version_raise)
+        pack_errors = pack_unique_files_validator.validate_pack_unique_files()
+        if pack_errors:
+            click.secho(pack_errors, fg="bright_red")
+            return False
+
+        return True
+
+    def validate_modified_files(self, modified_files):
+        click.secho(f'\n================= Running validation on modified files =================',
+                    fg="bright_cyan")
+        valid_files = set()
+        for file_path in modified_files:
+            # handle renamed files
+            if isinstance(file_path, tuple):
+                old_file_path = file_path[0]
+                file_path = file_path[1]
+
+            else:
+                old_file_path = None
+
+            pack_name = get_pack_name(file_path)
+            valid_files.add(self.run_validations_on_file(file_path, self.get_error_ignore_list(pack_name),
+                                                         is_modified=True, old_file_path=old_file_path))
+        return all(valid_files)
+
+    def validate_added_files(self, added_files, modified_files):
+        click.secho(f'\n================= Running validation on newly added files =================',
+                    fg="bright_cyan")
+
+        valid_files = set()
+        for file_path in added_files:
+            pack_name = get_pack_name(file_path)
+            valid_files.add(self.run_validations_on_file(file_path, self.get_error_ignore_list(pack_name),
+                                                         is_modified=False, modified_files=modified_files,
+                                                         added_files=added_files))
+        return all(valid_files)
+
+    def validate_changed_packs_unique_files(self, modified_files, added_files, changed_meta_files):
+        click.secho(f'\n================= Running validation on changed pack unique files =================',
+                    fg="bright_cyan")
+        valid_pack_files = set()
+
+        added_packs = get_pack_names_from_files(added_files)
+        modified_packs = get_pack_names_from_files(modified_files)
+
+        # modified packs (where the change is not test-playbook, test-script, readme or release notes)
+        # and packs where the meta file changed should have their version raised
+        modified_packs_that_should_have_version_raised = get_pack_names_from_files(modified_files, skip_file_types={
+            FileType.RELEASE_NOTES, FileType.README, FileType.TEST_PLAYBOOK, FileType.TEST_SCRIPT
+        })
+
+        packs_that_should_have_version_raised = get_pack_names_from_files(changed_meta_files).union(
+            modified_packs_that_should_have_version_raised)
+
+        changed_packs = modified_packs.union(added_packs).union(packs_that_should_have_version_raised)
```

Comment:
i don't really get all of these unions, why do we need them all

Reply:
all the changed_packs are packs that have changes in their files regardless if they are added files, modified files or packs that have their metadata file changed - this union is simply adding all of them